### PR TITLE
MATX-198 - Extracted code common between MaterialXSurfaceOverride and MaterialXTextureOverride to MaterialXShadingNodeImpl.

### PR DESCRIPTION
**Pull request by ppenenko**
The surface node behaves similarly to the texture node now. When trying to connect its output to the surface shader input on an SG, Maya asserts on shader compilation - something to investigate next.

### CODE REVIEW

**Comment by ppenenko** [comment]
Actually @bernardkwok now that we've decided to get rid of TestFileNodeOverride and the override classes become really trivial, I think it will be cleaner to put all of these classes in a single pair of header and source files.

File: source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+#include <maya/MShaderManager.h>
+#include <maya/MTextureManager.h>
+#include <maya/MPxShadingNodeOverride.h>
+#include <maya/MPxSurfaceShadingNodeOverride.h>
+
+#include <fstream>
+
+namespace
+{
+    // This should be a shared utility
```
Comment (bernardkwok) [inline source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp]: Minor: coding convention. We don't indent after a namespace declaration. Also should left align arguments in methods.

File: source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+    // Bind environment lighting
+    // TODO: These should be options
+    std::string envRadiancePath = "san_giuseppe_bridge.hdr";
+    std::string envIrradiancePath = "san_giuseppe_bridge_diffuse.hdr";
+    ::bindEnvironmentLighting(shader, globals, parameterList, imageSearchPath, 
+                            envRadiancePath, envIrradiancePath);
+
+    MaterialX::DocumentPtr document = node->materialXData->getDocument();
+	const MaterialX::StringMap& inputs = node->materialXData->getPathInputMap();
```
Comment (bernardkwok) [inline source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp]: Looks like all your tabbing is still messed up. Should be 4 spaces, no hard-tabs.

File: source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#ifndef MaterialXShadingNodeImpl_H
+#define MaterialXShadingNodeImpl_H
+
+#include "../OGSXMLFragmentWrapper.h"
```
Comment (bernardkwok) [inline source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.h]: We set up include cmake paths so should use <MaterialXContrib/...>.

File: source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp (hunk 2 above)
Comment (ppenenko) [reply source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp]: This file was copied from MaterialXTextureOverride, the tabs are left over from there. I'll clean up.

**Pull request merged**